alacritty_terminal/src/selection.rs

```diff
@@ -165,179 +172,165 @@ impl Selection {
         }
     }
 
-    pub fn to_span<T>(&self, term: &Term<T>) -> Option<Span> {
-        // Get both sides of the selection
-        let (mut start, mut end) = match *self {
-            Selection::Simple { ref region } | Selection::Block { ref region } => {
-                (region.start.point, region.end.point)
-            },
-            Selection::Semantic { ref region } | Selection::Lines { ref region } => {
-                (region.start, region.end)
-            },
-        };
+    /// Convert selection to grid coordinates.
+    pub fn to_range<T>(&self, term: &Term<T>) -> Option<SelectionRange> {
+        let grid = term.grid();
+        let num_cols = grid.num_cols();
 
-        // Order the start/end
-        let needs_swap = Selection::points_need_swap(start, end);
-        if needs_swap {
-            std::mem::swap(&mut start, &mut end);
-        }
+        // Get selection boundaries
+        let points = self.points();
+        let (mut start, mut end) = (*points.0, *points.1);
 
-        // Clamp to visible region in grid/normal
-        let num_cols = term.dimensions().col;
-        let num_lines = term.dimensions().line.0 as isize;
-        let (start, end) = Selection::grid_clamp(start, end, num_lines, num_cols)?;
+        // Get selection sides, falling back to `Side::Left` if it will not be used
+        let sides = self.sides().unwrap_or((&Side::Left, &Side::Left));
+        let (mut start_side, mut end_side) = (*sides.0, *sides.1);
 
-        let span = match *self {
-            Selection::Simple { ref region } => {
-                let (start_side, end_side) = if needs_swap {
-                    (region.end.side, region.start.side)
-                } else {
-                    (region.start.side, region.end.side)
-                };
+        // Order start above the end
+        if Self::points_need_swap(start, end) {
+            mem::swap(&mut start_side, &mut end_side);
+            mem::swap(&mut start, &mut end);
+        }
 
-                self.span_simple(term, start, end, start_side, end_side)
-            },
-            Selection::Block { ref region } => {
-                let (start_side, end_side) = if needs_swap {
-                    (region.end.side, region.start.side)
-                } else {
-                    (region.start.side, region.end.side)
-                };
+        // Clamp to inside the grid buffer
+        let is_block = self.is_block();
+        Self::grid_clamp(&mut start, &mut end, &mut start_side, is_block, grid.len()).ok()?;
```

Comment:
I have intentionally extracted `is_block` here to prevent `.ok()?;` from being moved to the next line by rustfmt. I think having two separate stataments makes things more readable than having a single line that is broken across two.

Reply:
The thing is that it's one line now, since you have a useless `mut` ;- ).

Reply:
Oh, I didn't notice that in the diff, you're entirely right.

Kinda surprising though that no Rust tool was able to lint that. It seems like that would be a nice thing for clippy to mention, that you don't need to pass a mutable reference.

Reply:
clippy failed on it.

Reply:
You're absolutely right, how could I have doubted `clippy` first before myself. Blasphemous!